addTimeShift_function.R

```diff
@@ -0,0 +1,69 @@
+require(lubridate)
+require(dplyr)
+
+#' Add column of time-shifted shifted metrics
+#' 
+#' This function takes a dataframe with a date column and a metric column,
+#' along with any other dimensions in the dataset, and creates a new
+#' metric column time-shifted by the parameters specified. E.g. clicks for
+#' this row one year ago or one week ago.
+#' 
+#' @param data.in The dataframe containing a date and metric
+#' @param date The date column in the dataframe
+#' @param n.per The period of time to shift by (day, week, month, year)
+#' @param n.shift The number of period shifts to apply
+#' @export
+
+addTimeShift <- function(data.in,
+                         metric,
+                         date.col = "date",
+                         time.shift = 1,
+                         shift.per = 364) {
```

Comment:
I'm a little confused by these shift things.

* Why is 364 the default? If you're shifting by a year, wouldn't you use 365?
* What is time.shift? If it's supposed to be the same thing as `n.shift` in the function comments, does `time.shift = n` lead to the same result as multiplying `shift.per` by n?